src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -728,6 +728,691 @@ public object WProp
         }
 
         [Fact]
+        public void TestFieldAttributeOnPropertyInCSharp7_2()
+        {
+            string source = @"
+public class A : System.Attribute
+{
+}
+public class Test
+{
+    [field: System.Obsolete]
+    [field: A]
+    public int P { get; set; }
+
+    [field: System.Obsolete(""obsolete"", error: true)]
+    public int P2 { get; }
+}
+";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (7,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: System.Obsolete]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, "[field: System.Obsolete]").WithArguments("7.2", "7.3").WithLocation(7, 5),
+                // (8,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: A]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, "[field: A]").WithArguments("7.2", "7.3").WithLocation(8, 5),
+                // (11,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: System.Obsolete("obsolete", error: true)]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, @"[field: System.Obsolete(""obsolete"", error: true)]").WithArguments("7.2", "7.3").WithLocation(11, 5)
+                );
+        }
+
+        [Fact]
+        public void TestFieldAttributesOnAutoProperty()
```

Comment:
>TestFieldAttributesOnAutoProperty [](start = 20, length = 33)

It would be good to test attribute targets other than field or property used on an auto-property.

Reply:
It looks like there is a test like this in AttributeTests_Locations.cs.

---
In reply to: [148176418](https://github.com/dotnet/roslyn/pull/22664#discussion_r148176418) [](ancestors = 148176418)